### PR TITLE
feat(dsa): add SM89 FP8 paged indexer operator

### PR DESCRIPTION
**Pull request by mochgolf**
## Motivation

Ada/SM89 cannot use the SM90-only fast DSA indexer paths. GLM DSA also uses a
different raw paged index-cache layout from the DeepSeek V4 work in #28620 and
#25470: each page contains 64 E4M3 key vectors of dimension 128 followed by one
FP32 dynamic scale per token (`8448` bytes total).

This PR adds the exact paged-logits primitive for that layout. It is not a
dense-attention fallback and does not change any runtime default or backend
selection by itself.

## Modifications

- Add a Triton paged FP8 index-logits operator under the unified
  `sglang.kernels.ops.attention.dsa` namespace introduced by #29630 and #30792.
- Register the operator lazily as
  `attention.sm89_paged_fp8_index_logits`, with an exact SM89 capability
  requirement using the current set-valued `KernelSpec.capabilities` API.
- Support page size 64, 32 query heads, and head dimension 128.
- Apply ReLU per head before the signed FP32 head-weight reduction, then apply
  each token's packed FP32 dynamic scale.
- Read logical-to-physical page mappings directly and leave invalid pages and
  tokens at `-inf`.
- Validate dtype, shape, contiguity, device, and page-width contracts at the
  Python boundary.
- Keep the launch graph-safe: the kernel path performs no host reads or tensor
  value-dependent Python control flow.
- Add numerical, contract, non-contiguous page-table, masking, CUDA Graph
  replay, and CPU-only registry tests.

The scope is intentionally operator-only. A follow-up will connect it to the
GLM DSA indexer selector together with the SM89 sparse MLA backend; until then
the operator is not selected by serving code.

## Accuracy Tests

Hardware: RTX 4090 48GB mod (SM89), Torch 2.11.0+cu129.

```bash
PYTHONPATH=python python -m pytest \
  test/registered/kernels/test_sm89_paged_indexer.py -q
```

Result: `25 passed, 5 warnings in 6.30s`.

The tests compare logits and sorted top-k indices against an eager FP32
reference, including signed head weights, non-unit per-token scales, invalid
physical pages, partial final pages, and stride-2 page-table views. CUDA Graph
tests replay short-to-long and long-to-short sequence lengths while checking
stable input pointers and exact masking.

CPU-only namespace and registry validation:

```bash
PYTHONPATH=python python -m pytest \
  test/registered/kernels/test_kernels_namespace.py -q
```

Result: `51 passed, 3 warnings in 10.44s`.

Downstream integration validation, not claimed as functionality of this PR
alone, was also completed on 2x RTX 4090 48GB mod GPUs with GLM-5.2-504B,
exact DSA, FP8 KV, TP=2, and a 262208-token KV budget. A cold 200006-token
prompt plus exact 128 output tokens measured 105.785 tok/s prefill and 9.955
tok/s decode; three cached exact-128 runs had a 9.864 tok/s median and no
detected quality flags.
The stacked integration source is available on the
[`glm52-sm89-fp8-upstream`](https://github.com/mochgolf/sglang/tree/glm52-sm89-fp8-upstream)
branch.

## Speed Tests and Profiling

Standalone CUDA-event timing on one RTX 4090 48GB mod after warmup, batch size
1, all pages valid:

| Sequence length | Eager median | CUDA Graph replay median |
|---:|---:|---:|
| 32768 | 0.068 ms | 0.009 ms |
| 262144 | 0.068 ms | 0.021 ms |

Eager timing includes output allocation/initialization. Graph timing used 50
replays. There is no existing operator for this GLM layout on SM89, so this PR
does not claim a synthetic speedup ratio.

## Additional Checks

- Full repository pre-commit hooks on `origin/main..HEAD`
- `scripts/ci/check_registered_tests.py`
- `scripts/ci/check_no_registered_tests_in_package.py`
- `git diff --check`

## Checklist

- [x] Rebase onto current upstream main.
- [x] Use the unified `sglang.kernels.ops` public namespace and registry.
- [x] Format the changed files according to the repository pre-commit versions.
- [x] Add focused unit, registry, and CUDA Graph tests.
- [x] Provide accuracy and standalone latency results.
- [x] No user-facing configuration is introduced here; serving documentation
      will accompany the integration follow-up.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #29725364773](https://github.com/sgl-project/sglang/actions/runs/29725364773)
Latest PR Test (Extra): :hourglass_flowing_sand: [Run #29725364644](https://github.com/sgl-project/sglang/actions/runs/29725364644)

### CODE REVIEW

**Comment by mochgolf** [comment]
Rebased onto current `main` (`a61482134`) and adapted this operator-only A1 PR
to the unified `sglang.kernels.ops` namespace and `KernelSpec` registry from
#29630 / #30792. The serving selector remains unchanged.

Fresh local validation on an RTX 4090 48GB mod:

- GPU numerical, contract, and CUDA Graph tests: `25 passed`
- CPU-only namespace/registry tests: `12 passed`
- Full pre-commit on `origin/main..HEAD`: passed
- Upstream Lint workflow: passed

@BBuf, could you review the kernel namespace / registration placement?
@Fridge003, could you review the DSA math and packed page-64 FP8 layout
contract? If the operator-only scope looks appropriate, please also add the
`run-ci` label. Follow-up selector and sparse-MLA integration PRs will remain
separate and serial.

**Comment by mochgolf** [comment]
Follow-up after four days: rebased this operator-only A1 onto current `main`
(`2f14d6c6f`); new head is `074294e5b`.

Upstream changed `KernelSpec` capability metadata and rewrote the registry tests,
so this update uses the current set-valued `capabilities` API and current pytest
registry suite. The paged FP8 operator and its GPU test are byte-identical to the
prior head; serving selection remains unchanged.

Fresh validation on RTX 4090 48GB mod:
- GPU numerical/contract/CUDA Graph: `25 passed`
- CPU namespace/registry: `51 passed`
- full pre-commit: passed
- upstream Lint: passed

@BBuf @Fridge003, when convenient, could you review this operator-only scope and
add `run-ci` if appropriate?
